internal/artifacts/echoes/echoes.go

```diff
@@ -0,0 +1,57 @@
+package echoes
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterSetFunc("echoes of an offering", New)
+	core.RegisterSetFunc("echoesofanoffering", New)
+	core.RegisterSetFunc("echoes", New)
+}
+
+func New(c core.Character, s *core.Core, count int, params map[string]int) {
+	prob := 0.36
+	probicd := 0
+	if count >= 2 {
+		m := make([]float64, core.EndStatType)
+		m[core.ATKP] = 0.18
+		c.AddMod(core.CharStatMod{
+			Key: "echoes-2pc",
+			Amount: func() ([]float64, bool) {
+				return m, true
+			},
+			Expiry: -1,
+		})
+	}
+
+	if count >= 4 {
+		s.Events.Subscribe(core.OnAttackWillLand, func(args ...interface{}) bool {
+			if s.ActiveChar != c.CharIndex() {
+				return false
+			}
+			if args[1].(*core.AttackEvent).Info.AttackTag != core.AttackTagNormal {
+				return false
+			}
+			atk := args[1].(*core.AttackEvent)
+			if s.Rand.Float64() < prob {
+				snap := c.Snapshot(&atk.Info)
```

Comment:
i THINK it might be better to use the stats from the attack rather than creating another snapshot. we'll have to see when it goes live but i feel it's more likely to use existing stats than snapshot again.